app/code/Magento/CatalogGraphQl/Model/Resolver/Products.php

```diff
@@ -94,4 +81,29 @@ public function resolve(
 
         return $data;
     }
+
+    /**
+     * Validate input arguments
+     *
+     * @param array $args
+     * @throws GraphQlAuthorizationException
+     * @throws GraphQlInputException
+     */
+    private function validateInput(array $args)
+    {
+        if (isset($args['searchAllowed']) && $args['searchAllowed'] === false) {
+            throw new GraphQlAuthorizationException(__('Product search has been disabled.'));
```

Comment:
did we decide that this was preferable against no results?
shouldn't we have a storeConfig for it?

Reply:
When the customer group is denied searching the catalog, luma behaves in such a way that it would redirect to 404 page if the search results controller is hit (and that is if one knows or has Url to the search results previously saved or bookmarked).
So, luma does NOT land on a search results page saying that we do not have any results.

I thought this is the closest to that behavior, although, it gives information that luma does not: that the product search has been disabled.

I assume adding a store config about it, whether to throw an exception, or, display empty set of results, should be elegant approach.